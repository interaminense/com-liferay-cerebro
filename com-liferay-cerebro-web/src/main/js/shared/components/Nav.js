import getCN from 'classnames';
import Component, {Config} from 'metal-jsx';

const DISPLAYS = ['pills', 'tabs', 'underline'];

class Item extends Component {
	render() {
		const {active, children, href, onClick} = this.props;

		let content = children;

		if (href) {
			content = (
				<a
					class={getCN('nav-link', {active})}
					href={href}
					onClick={onClick}
					role="tab"
				>
					{children}
				</a>
			);
		}

		return (
			<li
				{...this.otherProps()}
				class="nav-item"
				key={href}
				role="presentation"
			>
				{content}
			</li>
		);
	}
}

Item.STATE = {
	active: Config.bool(),
	href: Config.string(),
	label: Config.string()
};

class Nav extends Component {
	render() {
		const {context: {navBar}, props: {children, display, vertical}} = this;

		const classes = getCN('nav-root', {
			'flex-column': vertical,
			nav: !navBar,
			'navbar-nav': navBar,
			[`nav-${display}`]: display
		});

		return (
			<ul class={classes} role="tablist">
				{children}
			</ul>
		);
	}
}

Nav.PROPS = {
	display: Config.oneOf(DISPLAYS),
	vertical: Config.bool()
};

Nav.Item = Item;

export default Nav;