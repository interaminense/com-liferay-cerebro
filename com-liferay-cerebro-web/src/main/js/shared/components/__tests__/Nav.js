import Nav from '../Nav';

describe('Nav', () => {
	let component;

	const items = [
		<Nav.Item active href="#" key={1}>
			{'foo'}
		</Nav.Item>,
		<Nav.Item key={2}>{'bar'}</Nav.Item>,
		<Nav.Item key={3}>{'baz'}</Nav.Item>
	];

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Nav();

		expect(component).toMatchSnapshot();
	});

	it('should render vertically', () => {
		component = new Nav({
			vertical: true
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with items', () => {
		component = new Nav({
			children: items
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with items as pills', () => {
		component = new Nav({
			children: items,
			display: 'pills'
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with items as tabs', () => {
		component = new Nav({
			children: items,
			display: 'tabs'
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with underline class', () => {
		component = new Nav({
			children: items,
			display: 'underline'
		});

		expect(component).toMatchSnapshot();
	});
});