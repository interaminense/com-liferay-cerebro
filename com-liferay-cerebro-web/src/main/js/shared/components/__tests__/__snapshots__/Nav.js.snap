// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render 1`] = `<ul class="nav-root nav" role="tablist"></ul>`;

exports[`Nav should render vertically 1`] = `<ul class="nav-root flex-column nav" role="tablist"></ul>`;

exports[`Nav should render with items 1`] = `
<ul class="nav-root nav" role="tablist">
  <li class="nav-item" role="presentation" active="" href="#">
    <a class="nav-link active" href="#" role="tab">foo</a>
  </li>
  <li class="nav-item" role="presentation">bar</li>
  <li class="nav-item" role="presentation">baz</li>
</ul>
`;

exports[`Nav should render with items as pills 1`] = `
<ul class="nav-root nav nav-pills" role="tablist">
  <li class="nav-item" role="presentation" active="" href="#">
    <a class="nav-link active" href="#" role="tab">foo</a>
  </li>
  <li class="nav-item" role="presentation">bar</li>
  <li class="nav-item" role="presentation">baz</li>
</ul>
`;

exports[`Nav should render with items as tabs 1`] = `
<ul class="nav-root nav nav-tabs" role="tablist">
  <li class="nav-item" role="presentation" active="" href="#">
    <a class="nav-link active" href="#" role="tab">foo</a>
  </li>
  <li class="nav-item" role="presentation">bar</li>
  <li class="nav-item" role="presentation">baz</li>
</ul>
`;

exports[`Nav should render with underline class 1`] = `
<ul class="nav-root nav nav-underline" role="tablist">
  <li class="nav-item" role="presentation" active="" href="#">
    <a class="nav-link active" href="#" role="tab">foo</a>
  </li>
  <li class="nav-item" role="presentation">bar</li>
  <li class="nav-item" role="presentation">baz</li>
</ul>
`;
